tools/clang/lib/SPIRV/AstTypeProbe.cpp

```diff
@@ -735,13 +730,30 @@ bool isSameScalarOrVecType(QualType type1, QualType type2) {
       return canTreatAsSameScalarType(scalarType1, scalarType2);
   }
 
-  { // Vector types
+  do { // Vector types
     QualType elemType1 = {}, elemType2 = {};
     uint32_t count1 = {}, count2 = {};
-    if (isVectorType(type1, &elemType1, &count1) &&
-        isVectorType(type2, &elemType2, &count2))
-      return count1 == count2 && canTreatAsSameScalarType(elemType1, elemType2);
-  }
+    if (!isVectorType(type1, &elemType1, &count1) ||
+        !isVectorType(type2, &elemType2, &count2))
+      break;
```

Comment:
Can you remove the `do { ... } while (false);` and replace the `break` with `return false` instead?

Reply:
Yes, that's better!